pkg/disk/disk.go

```diff
@@ -58,6 +59,54 @@ const (
 	XBootLDRPartitionGUID = "BC13C2FF-59E6-4262-A352-B275FD6F7172"
 )
 
+// FSType is the filesystem type enum.
+//
+// There should always be one value for each filesystem type supported by
+// osbuild stages (stages/org.osbuild.mkfs.*) and the unset/none value.
+type FSType uint64
+
+const (
+	FS_NONE FSType = iota
+	FS_VFAT
+	FS_EXT4
+	FS_XFS
+	FS_BTRFS
+)
+
+func (f FSType) String() string {
+	switch f {
+	case FS_NONE:
+		return ""
+	case FS_VFAT:
+		return "vfat"
+	case FS_EXT4:
+		return "ext4"
+	case FS_XFS:
+		return "xfs"
+	case FS_BTRFS:
+		return "btrfs"
+	default:
+		panic(fmt.Sprintf("unknown or unsupported filesystem type with enum value %d", f))
+	}
+}
+
+func FSTypeFromString(s string) (FSType, error) {
```

Comment:
(nitpick) maybe just `NewFSType(s) (fstype, error)` ?

Reply:
Yeah, `New` is nicer.  I was basing this off the `Arch` enum where we have an `arch.FromString(str)` function, but maybe we can change that too.

Reply:
Fwiw, it's a nitpick, fine to leave and get back to later (or ignore), especially if there is already a different pattern.